packages/components/src/internal/components/assay/AssayImportSubMenuItem.tsx

```diff
@@ -1,25 +1,13 @@
 import React, { FC, useMemo } from 'react';
 import { MenuItem, OverlayTrigger, Popover } from 'react-bootstrap';
 
-import {
-    getImportItemsForAssayDefinitions,
-    InjectedAssayModel,
-    SubMenuItem,
-    SubMenuItemProps,
-    QueryGridModel,
-    withAssayModels,
-    QueryModel,
-} from '../../..';
+import { InjectedAssayModel, SubMenuItem, SubMenuItemProps, withAssayModels, QueryModel } from '../../..';
 import { MAX_EDITABLE_GRID_ROWS } from '../../constants';
 
-import { getImportItemsForAssayDefinitionsQM } from './actions';
+import { getImportItemsForAssayDefinitions } from './actions';
 
 interface Props extends SubMenuItemProps {
     isLoaded?: boolean;
-    /**
-     * @deprecated: Use QueryModel instead.
-     */
-    model?: QueryGridModel;
     queryModel?: QueryModel;
```

Comment:
Now that there is only one configuration `queryModel` should be marked as required. Additionally, the following logic within the component can assume a `queryModel` is available.

Reply:
Yes, I think that the `queryModel` property can new be required, but I think we need to be a little more defensive in the usages of that property within this component. For example, in the LKB usages from `CreateButton.tsx` line 115, it passes the prop as `queryModel={isSamplesContext && selectionQueryModel}`. So I think the usage in this component to get `selectedCount` needs to add a null/undefined check.